Small modifications based on code review of asynchronous prefetching

Dr15Jones opened the pull request:
Although the first argument to the constructor is an lvalue reference, it is still required to use std::move to get the benefits.

Dr15Jones committed:
Use std::move in SharedResourcesAcquirer constructor

Although the first argument to the constructor is an lvalue reference, it is still required to use std::move to get the benefits.

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones (Chris Jones) for CMSSW_8_1_X.

It involves the following packages:

FWCore/Framework


@cmsbuild, @smuzaffar, @Dr15Jones, @davidlange6 can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @wddgit, @wmtan this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

Dr15Jones commented:
please test

Dr15Jones commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

Dr15Jones committed:
Added additional comments

cmsbuild commented:
-1

Tested at: 1bf8031b293777a50963c7937deea10e033e2c75

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15456/14507/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test testJetMETCorrectionsType1MET had ERRORS

Dr15Jones commented:
The unit test failure is unrelated to this pull request.

Dr15Jones committed:
Properly handle non CMS exceptions in Worker::runModuleAfterAsyncPrefetch

Moved code into the convertException::wrap just in case a non-CMS exception was thrown.

Dr15Jones committed:
Use sentry to guarantee ActivityRegistry signal happens

Use an std::unique_ptr to guarantee that at then end of the code block the postModuleEventPrefetchSignal will be emitted, even if an exception happens.

cmsbuild commented:
Pull request #15456 was updated. @cmsbuild, @smuzaffar, @Dr15Jones, @davidlange6 can you please check and sign again.

Dr15Jones commented:
I accidently pushed additional commits to the same branch. Now it happens that all commits are based off the same code review, so it isn't really neccesary for them to be separate.

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

Dr15Jones commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15456/14507/summary.html

cmsbuild commented:
-1

Tested at: b80eba3311efdd538a694624f9ca121182750d86

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15456/14514/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test testJetMETCorrectionsType1MET had ERRORS

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15456/14514/summary.html

Dr15Jones commented:
The unit test failure is not related to this pull request.
